### ISSUE TITLE
High CPU Usage

### ISSUE THREAD

**Issue by nighthawk469** [nbolar/PlayStatus#19]
I noticed PlayStatus on MacOS 10.15.7 was taking sitting at 17% CPU usage and causing my fans to spin, even though there wasn't any music playing in Apple Music. I wonder if you had any thoughts about that. Thanks

**Comment by nighthawk469** [comment]
I'll play around with it and see if it is caused by the "scrollable text" option.

**Comment by nbolar** [comment]
Hey @nighthawk469, thanks for letting me know. Hmm okay off the top of my head I'm not sure why it's that high but I'll monitor it and try making some changes. I'll keep you updated should anything change.

**Comment by nbolar** [comment]
So I made some optimizations and it looks like it's using a lot less of the CPU now. I'll keep monitoring it for a few more days and if all goes well, then I'll push an update early next week.

**Comment by nighthawk469** [comment]
Thats great, looking forward to it. I wonder if the performance is partly due to using AppleScript instead of Swift code.

**Comment by nbolar** [comment]
Yes, that definitely does cause a slight performance hit. Apple Script is fairly efficient though. However, since I don't have system level access to the macOS Music Player, I rely on timers constantly running in the background to notify me of changes using Apple Script. With that being said, I've reduced the frequency with which these timers are called.

**Comment by nbolar** [comment]
I've noticed an issue specifically when using the scrollable text feature. For most of the time I spent testing, when the titles are scrollable, the CPU usage is fairly low. But for some reason when the memory usage spikes so does the CPU usage even on idle. It could very well be the way the scrollable text feature was implemented. Unfortunately it's a package that I imported and I have not developed it myself. So I'm unsure where to begin debugging it. As it stands right now, there are two options, if the CPU/memory usage is too high, the app should be restarted or turn off the scrollable text option. I will release what I have done so far later on today but will keep working on the scrollable fix whenever I get the time. Apologies for this.
